setup_db.sh

```diff
@@ -0,0 +1,124 @@
+
+
+# Function to check if redis CR is available
+check_redis_available() {
+    local message=$(oc get redis "$1" -o jsonpath="{.status.message}")
+    if [[ "$message" == *"redis deployment available"* ]]; then
+        return 0  # Redis CR is available
+    else
+        return 1  # Redis CR is not available
+    fi
+}
+
+# Function to check if redis CR phase is complete
+check_redis_complete() {
+    local phase=$(oc get redis "$1" -o jsonpath="{.status.phase}")
+    if [[ "$phase" == "complete" ]]; then
+        return 0  # Redis CR phase is complete
+    else
+        return 1  # Redis CR phase is not complete
+    fi
+}
+
+# Function to check if postgres CR is available
+check_postgres_available() {
+    local message=$(oc get postgres example-postgres -o jsonpath="{.status.message}")
+    if [[ "$message" == *"creation successful"* ]]; then
+        return 0  # Postgres CR is available
+    else
+        return 1  # Postgres CR is not available
+    fi
+}
+
+# Function to check if postgres CR phase is complete
+check_postgres_complete() {
+    local phase=$(oc get postgres example-postgres -o jsonpath="{.status.phase}")
+    if [[ "$phase" == "complete" ]]; then
+        return 0  # Postgres CR phase is complete
+    else
+        return 1  # Postgres CR phase is not complete
+    fi
+}
+
+
+git clone git@github.com:integr8ly/cloud-resource-operator.git
+cd cloud-resource-operator
+make cluster/prepare
+REDIS_NAME=redis-storage make cluster/seed/redis
+REDIS_NAME=redis-queue make cluster/seed/redis
+REDIS_NAME=redis-system make cluster/seed/redis
+make cluster/seed/postgres
+
+RECTIME=30 WATCH_NAMESPACE=cloud-resource-operator go run ./main.go &>/dev/null &
+
+# Check postgres is completed
+while true; do
+    if check_postgres_available && check_postgres_complete; then
+        echo "Both postgres conditions are satisfied for example-postgres. Proceeding with the script."
+        break  # Exit the loop when both conditions are satisfied
+    else
+        echo "One or both postgres conditions are not satisfied for example-postgres. Retrying in 10 seconds."
+        sleep 10  # Wait for 10 seconds before retrying
+    fi
+done
+# Check all redis are completed
+redis_variables=("redis-storage" "redis-queue" "redis-system")
+
+for var in "${redis_variables[@]}"; do
+    while true; do
+        if check_redis_available "$var" && check_redis_complete "$var"; then
+            echo "Both redis conditions are satisfied for $var. Proceeding with the script."
+            break  # Exit the loop when both conditions are satisfied
+        else
+            echo "One or both conditions are not satisfied for $var. Retrying in 10 seconds."
+            sleep 10  # Wait for 10 seconds before retrying
+        fi
+    done
+done
+
+                                                      
+# Kill the operator process
+pkill -f "main.go"
+# delete the cloud-resources-operator dir
+cd ..
+rm -rf cloud-resources-operator
```

Comment:
Not sure why but this command isn't deleting the `cloud-resources-operator` dir. Then if you try to run the script again it fails because it can't clone into the existing non-empty directory.

Reply:
I can think of no reason why `rm -rf` wouldn't work :thinking: 
Ah it should be `cloud-resource-operator`

Did see sometimes the port 8383 wasn't getting released after the main.go process was killed but it's a common enough issue with running operators locally and didn't try and fix it here.